src/connect/context/ItemsContext.tsx

```diff
@@ -0,0 +1,41 @@
+import { TreeItem } from '@/connect/components/tree-view-item';
+import React, { useContext, useState } from 'react';
+
+interface TreeItemContext {
+    items: TreeItem[];
+    setItems: React.Dispatch<React.SetStateAction<TreeItem[]>>;
+}
+
+const defaultTreeItemContextValue: TreeItemContext = {
+    items: [],
+    setItems: () => {},
+};
+
+export const ItemsContext = React.createContext<TreeItemContext>(
+    defaultTreeItemContextValue,
+);
+
+export interface ItemsContextProviderProps {
+    children?: React.ReactNode;
+    items?: TreeItem[];
+}
+
+export const ItemsContextProvider: React.FC<ItemsContextProviderProps> = ({
```

Comment:
This is a base implementation of the ItemsContextProvider. In a next commit I'll add support to handle UI specific state, general TreeItem state, and merge both states into the TreeItem flat list

Reply:
this is excellent work